src/MyConnection.h

```diff
@@ -18,10 +18,15 @@ class MyConnection : boost::noncopyable {
 
 public:
 
-  MyConnection(std::string host, std::string user, std::string password,
-               std::string db, unsigned int port, std::string unix_socket,
-               unsigned long client_flag, std::string groups,
-               std::string default_file) :
+  MyConnection(Rcpp::Nullable < std::string > host,
```

Comment:
Yes, `const&` are preferable.

I thought about the first option: Wait for Rcpp CRAN release before pulling this into RMySQL. You could also do the changes that depend on an updated Rcpp in another branch, so that people can use this PR if needed, without having to use the dev version of Rcpp.

Thanks for your help!